rpc/wrtc_base_channel.go

```diff
@@ -108,8 +109,13 @@ func newBaseChannel(
 					"conn_state", connectionState.String(),
 				)
 				doPeerDone()
+			case webrtc.ICEConnectionStateConnected:
+				if onICEConnected != nil {
+					onICEConnected()
```

Comment:
I had to do some funky `onICEConnected` callback logic here, as defining an `OnICEConnectionStateChange` handler after `newBaseChannel` ran would've overwritten the existing one (see the `connStateChanged` anonymous function this line is in). Only one specified handler is allowed.

Reply:
it took me a second to understand where the overwrite was happening: https://github.com/viamrobotics/goutils/blob/377d0dcf55f18e19ffb46d53363b5a02a8e6aa71/rpc/wrtc_base_channel.go#L134

makes sense now